base/libdl.jl

```diff
@@ -330,10 +330,9 @@ libfoo = LazyLibrary(LazyLibraryPath(prefix, "lib/libfoo.so.1.2.3"))
 ```
 """
 struct LazyLibraryPath
-    pieces::Vector
-    LazyLibraryPath(pieces::Vector) = new(pieces)
+    pieces::Tuple{Vararg{Any}}
```

Comment:
Why not just `Tuple`, they're identically the same type after all:

```suggestion
    pieces::Tuple
```

Reply:
I think it's slightly clearer to write out the type explicitly, so that it's obvious this is a "`Vector{Any}`-like" Tuple, rather than just an sloppily-annotated-but-in-practice-more-specific `Tuple{...}`